### ISSUE TITLE
`onMount` cast on a computed store disables the whole chain of reactivity

### ISSUE THREAD

**Issue by dkzlv** [nanostores/nanostores#96]
[Repro](https://codesandbox.io/s/nervous-estrela-ggq5z?file=/src/App.js).

<details>
<summary>Updated test (that fails) for issue #94</summary>

```typescript
test('is compatible with onMount', () => {
  const initialStoreValue = 1
  let store = atom<number>(initialStoreValue)
  let deferrer = computed(store, value => value * 2)

  let events = ''
  onMount(deferrer, () => {
    events += 'init '
    return () => {
      events += 'destroy '
    }
  })
  equal(events, '')

  let storeValue
  let unbind = deferrer.listen((value) => storeValue = value)
  clock.runAll()
  ok(deferrer.lc > 0)
  ok(storeValue === initialStoreValue * 2)
  ok(store.lc > 0)
  equal(events, 'init ')

  unbind()
  clock.runAll()
  equal(deferrer.lc, 0)
  equal(events, 'init destroy ')
})
```
</details>

I'll try to dig into this issue a little bit in the next few days, but I'll happily accept a fix made by someone else 😂

**Comment by ai** [comment]
Thanks for the test. I will try to look today.

**Comment by ai** [comment]
Note, that `listen` runs only on store changes. Since you didn’t change, `store` it was not run.

I changed test to `sunbscribe` which also run imminently.

**Comment by ai** [comment]
Seems like I found the reason. Additional `onMount` break the previous one (in `computed`). I will try to debug event system.

**Comment by ai** [comment]
Fixed d09eac6

**Comment by ai** [comment]
The fix was released in 0.5.10.
